retrier_test.go

```diff
@@ -112,6 +112,23 @@ func TestShouldGiveUp_WithMaxAttempts(t *testing.T) {
 	assert.Equal(t, 3, callcount)
 }
 
+func TestTimeout(t *testing.T) {
```

Comment:
this test super doesn't work, because the timeout function measures time inside the callback as well as time outside the callback (ie time spent waiting). The call count flakes between 92 and 93 before the timeout completes